packages/next/next-server/server/next-server.ts

```diff
@@ -1360,7 +1360,9 @@ export default class Server {
         // static paths always includes locale so make sure it's prefixed
         // with it
         !staticPaths.includes(
-          `${locale ? '/' + locale : ''}${resolvedUrlPathname}`
+          `${locale ? '/' + locale : ''}${decodeURIComponent(
```

Comment:
These changes should no longer be needed since this was updated in [https://github.com/vercel/next.js/pull/19135](https://github.com/vercel/next.js/pull/19135) please update to the latest canary and give it a try!

Reply:
I have confirmed that the latest version solves this problem. Thanks!